src/mpid/ch4/src/ch4r_win.h

```diff
@@ -1240,13 +1328,13 @@ static inline int MPIDI_CH4R_mpi_win_allocate(MPI_Aint size,
     if (mpi_errno != MPI_SUCCESS)
         goto fn_fail;
 
-    mpi_errno = MPIDI_CH4R_get_symmetric_heap(size, comm, &baseP, *win_ptr);
-
+    mpi_errno = MPIDI_CH4I_win_shm_alloc_impl(size, disp_unit, comm, &baseP, win_ptr);
     if (mpi_errno != MPI_SUCCESS)
         goto fn_fail;
 
     win = *win_ptr;
     win->base = baseP;
+    win->size = size;
```

Comment:
It was set in `MPIDI_CH4R_get_symmetric_heap`, which was inappropriate in my opinion. Because the symheap function should not set window specific attribute. Although I still pass `win` input in the new symheap function, it only updates symheap related parameters. 

In this commit, old `MPIDI_CH4R_get_symmetric_heap` is replaced with new `MPIDI_CH4I_win_shm_alloc_impl`, thus I need to add the size setting here.